src/lib.rs

```diff
@@ -339,23 +339,27 @@ fn create_http_server(
 fn cors_config(settings: &Settings) -> Cors {
   let self_origin = settings.get_protocol_and_hostname();
   let cors_origin_setting = settings.cors_origin();
+
+  // A default setting for either wildcard, or None
+  let cors_default = Cors::default()
+    .allow_any_origin()
+    .allow_any_method()
+    .allow_any_header()
+    .expose_any_header()
+    .max_age(3600);
+
   match (cors_origin_setting.clone(), cfg!(debug_assertions)) {
     (Some(origin), false) => {
       // Need to call send_wildcard() explicitly, passing this into allowed_origin() results in
       // error
-      if cors_origin_setting.as_deref() == Some("*") {
-        Cors::default().allow_any_origin().send_wildcard()
+      if origin == "*" {
+        cors_default
       } else {
         Cors::default()
           .allowed_origin(&origin)
           .allowed_origin(&self_origin)
```

Comment:
I have no idea if the `allow_any_method`, `allow_any_header` etc should also be added to this custom one.

Reply:
I wonder if this is even working without the extra settings.

> `Cors::default()`: A restrictive (security paranoid) set of defaults.
> 
> No allowed origins, methods, request headers or exposed headers. Credentials not supported. No max age (will use browser’s default).

I haven't tested setting this configuration yet, but it reads like all methods should be allowed at the very least.

I'm not sure about the headers, there are a number of headers exposed by browsers by default, but looking at requests in my browser i can see some additional headers being requested in cors preflight requests like authorization or content-type. any might be fine for this as well, sensitive headers like `set-cookie` on the response are protected anyway.

Reply:
In that case I'll add the other `allows`, except for origin, to the custom origin block.